File: js/method_example.js
import assert from "assert/strict"

const circle = {
  radius: 10,
  area: function () {
    return Math.PI * this.radius * this.radius
  },
  circumference: function () {
    return 2 * Math.PI * this.radius
  },
  expand: function (scale) {
    this.radius *= scale
  },
}

assert(circle.area() === 100 * Math.PI)
circle.expand(2)
assert(circle.circumference() === 40 * Math.PI)

File: dart/method_example.dart
import 'dart:math' show pi;

class Circle {
  static num radius = 10;
  static num get area => pi * radius * radius;
  static num get circumference => 2 * pi * radius;
  static void expand(scale) { radius *= scale; }
}

void main() {
  print(Circle.area);
  Circle.expand(2);
  print(Circle.circumference);
}
